src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -657,7 +657,7 @@ def save_as_table(
         )
         column_definition = re.sub(
             hidden_column_pattern,
-            lambda match: f"COL{match.group(1)}",
+            lambda match: f"${match.group(1)}",
```

Comment:
a correct fix for this will be to update `hidden_column_pattern = r"^\s*\$\d+\s*$"`